test/test_benchmark.py

```diff
@@ -921,6 +928,36 @@ def output_parser(output):
       return float(re.search(r'Total elapsed: ([\d\.]+)', output).group(1))
     self.do_benchmark('matrix_multiply', read_file(test_file('matrix_multiply.cpp')), 'Total elapsed:', output_parser=output_parser, shared_args=['-I' + test_file('benchmark')])
 
+  @non_core
+  def test_sizes(self):
+    if len(benchmarkers) != 1 or not isinstance(benchmarkers[0], EmscriptenBenchmarker):
+      self.skipTest('size benchmark only runs on emscripten')
+    benchmarkers[0].skip_bench = True
```

Comment:
Can we maybe replace the new `skip_bench` mechanism with a `skipTest()` that is called at the **end** of this function, after the work was done?

Reply:
I may be misunderstanding, but I think in that case `b.bench` in `do_benchmark` will still try to run and do multiple rounds of the benchmark.

Reply:
Hmm, I think you're right. This does need to work with the internals there. So I understand why this is needed to prevent `test_sizes` from being handled normally. But how do we prevent the sub-functions it calls (`test_primes` etc.) from also being treated in that way? That's what I find confusing I guess.

Separately, I think it would be nice if we could more clearly separate this function, which is a "meta-benchmark function" in a sense, from the normal benchmark functions. Perhaps instead of `@non_core` we could have a decorator `@meta` (or some better name) which implies both non-core (not run by default) and also the decorator sets `skip_bench`? The decorator would be a clearer indication of the special-ness of this function, rather than a line inside the function that sets `skip_bench`, is all I am thinking here.